service/controller/app/resource/appcatalog/create.go

```diff
@@ -10,7 +10,7 @@ import (
 
 // EnsureCreated gets the appCatalog CR specified in the provided app CR.
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
-	cr, err := key.ToCustomResource(obj)
+	cr, err := key.ToApp(obj)
```

Comment:
Package got flattened to support app and appcatalog CRs. `ToAppCatalog` already exists so aligning the naming.